### ISSUE TITLE
failing test on master

### ISSUE THREAD

**Issue by jonkiparsky** [bethlakshmi/GBE2#696]
I've nuked my local repo and rebuilt from github, so there's no local code issue, but I'm still getting a test failure in tests.scheduler.test_volunteer_management.test_scheduled_volunteer_opportunity_shows_day

editing the stack trace to just show our code, I get:

```
  File "/vagrant/expo/scheduler/views.py", line 864, in edit_event
    return edit_event_display(request, item)
  File "/vagrant/expo/scheduler/views.py", line 932, in edit_event_display
    return render(request, template, context)
...  # template handling
DoesNotExist: LocationItem matching query does not exist.
```

Betty says she's not getting this on her local.

**Comment by bethlakshmi** [comment]
When I can find a stopping point, I will nuke my own local and see if I can replicate.
I can tell you I've had other problems in around the code smell in Rooms before...

**Comment by jonkiparsky** [comment]
Much as it pains me to say it, don't worry about this until after the expo is done.

**Comment by bethlakshmi** [comment]
I ended up reprovisioning my vagrant this morning.  Still no failing test??
This definitely sounds like something that wants shoulder to shoulder work.

**Comment by bethlakshmi** [comment]
OK.  I m managed to fail in an interestingly similar way...! (woohoo??)

```
======================================================================
ERROR: tests.scheduler.test_volunteer_management.test_scheduled_volunteer_opportunity_shows_day
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/vagrant/expo/tests/scheduler/test_volunteer_management.py", line 57, in test_scheduled_volunteer_opportunity_shows_day
    response = edit_event(request, show_sevent.pk, event_type='Show')
  File "/vagrant/expo/scheduler/views.py", line 895, in edit_event
    return edit_event_display(request, item)
  File "/vagrant/expo/scheduler/views.py", line 963, in edit_event_display
    return render(request, template, context)
  File "/usr/local/lib/python2.7/dist-packages/django/shortcuts/__init__.py", line 53, in render
    return HttpResponse(loader.render_to_string(*args, **kwargs),
  File "/usr/local/lib/python2.7/dist-packages/django/template/loader.py", line 169, in render_to_string
    return t.render(context_instance)
  File "/usr/local/lib/python2.7/dist-packages/django/template/base.py", line 140, in render
    return self._render(context)
  File "/usr/local/lib/python2.7/dist-packages/django/test/utils.py", line 85, in instrumented_test_render
    return self.nodelist.render(context)
  File "/usr/local/lib/python2.7/dist-packages/django/template/base.py", line 840, in render
    bit = self.render_node(node, context)
  File "/usr/local/lib/python2.7/dist-packages/django/template/base.py", line 854, in render_node
    return node.render(context)
  File "/usr/local/lib/python2.7/dist-packages/django/template/loader_tags.py", line 123, in render
    return compiled_parent._render(context)
  File "/usr/local/lib/python2.7/dist-packages/django/test/utils.py", line 85, in instrumented_test_render
    return self.nodelist.render(context)
  File "/usr/local/lib/python2.7/dist-packages/django/template/base.py", line 840, in render
    bit = self.render_node(node, context)
  File "/usr/local/lib/python2.7/dist-packages/django/template/base.py", line 854, in render_node
    return node.render(context)
  File "/usr/local/lib/python2.7/dist-packages/classytags/core.py", line 106, in render
    return self.render_tag(context, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/sekizai/templatetags/sekizai_tags.py", line 79, in render_tag
    rendered_contents = nodelist.render(context)
  File "/usr/local/lib/python2.7/dist-packages/django/template/base.py", line 840, in render
    bit = self.render_node(node, context)
  File "/usr/local/lib/python2.7/dist-packages/django/template/base.py", line 854, in render_node
    return node.render(context)
  File "/usr/local/lib/python2.7/dist-packages/classytags/core.py", line 106, in render
    return self.render_tag(context, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/cms/templatetags/cms_tags.py", line 670, in render_tag
    rendered_contents = nodelist.render(context)
  File "/usr/local/lib/python2.7/dist-packages/django/template/base.py", line 840, in render
    bit = self.render_node(node, context)
  File "/usr/local/lib/python2.7/dist-packages/django/template/base.py", line 854, in render_node
    return node.render(context)
  File "/usr/local/lib/python2.7/dist-packages/django/template/loader_tags.py", line 62, in render
    result = block.nodelist.render(context)
  File "/usr/local/lib/python2.7/dist-packages/django/template/base.py", line 840, in render
    bit = self.render_node(node, context)
  File "/usr/local/lib/python2.7/dist-packages/django/template/base.py", line 854, in render_node
    return node.render(context)
  File "/usr/local/lib/python2.7/dist-packages/django/template/defaulttags.py", line 305, in render
    return nodelist.render(context)
  File "/usr/local/lib/python2.7/dist-packages/django/template/base.py", line 840, in render
    bit = self.render_node(node, context)
  File "/usr/local/lib/python2.7/dist-packages/django/template/base.py", line 854, in render_node
    return node.render(context)
  File "/usr/local/lib/python2.7/dist-packages/django/template/loader_tags.py", line 155, in render
    return self.render_template(self.template, context)
  File "/usr/local/lib/python2.7/dist-packages/django/template/loader_tags.py", line 137, in render_template
    output = template.render(context)
  File "/usr/local/lib/python2.7/dist-packages/django/template/base.py", line 140, in render
    return self._render(context)
  File "/usr/local/lib/python2.7/dist-packages/django/test/utils.py", line 85, in instrumented_test_render
    return self.nodelist.render(context)
  File "/usr/local/lib/python2.7/dist-packages/django/template/base.py", line 840, in render
    bit = self.render_node(node, context)
  File "/usr/local/lib/python2.7/dist-packages/django/template/base.py", line 854, in render_node
    return node.render(context)
  File "/usr/local/lib/python2.7/dist-packages/django/template/defaulttags.py", line 203, in render
    nodelist.append(node.render(context))
  File "/usr/local/lib/python2.7/dist-packages/django/template/base.py", line 897, in render
    return render_value_in_context(output, context)
  File "/usr/local/lib/python2.7/dist-packages/django/template/base.py", line 875, in render_value_in_context
    value = force_text(value)
  File "/usr/local/lib/python2.7/dist-packages/django/utils/encoding.py", line 100, in force_text
    s = s.__unicode__()
  File "/usr/local/lib/python2.7/dist-packages/django/forms/forms.py", line 425, in __str__
    return self.as_widget()
  File "/usr/local/lib/python2.7/dist-packages/django/forms/forms.py", line 475, in as_widget
    return widget.render(name, self.value(), attrs=attrs)
  File "/usr/local/lib/python2.7/dist-packages/django/forms/widgets.py", line 504, in render
    options = self.render_options(choices, [value])
  File "/usr/local/lib/python2.7/dist-packages/django/forms/widgets.py", line 535, in render_options
    output.append(self.render_option(selected_choices, option_value, option_label))
  File "/usr/local/lib/python2.7/dist-packages/django/forms/widgets.py", line 511, in render_option
    option_value = force_text(option_value)
  File "/usr/local/lib/python2.7/dist-packages/django/utils/encoding.py", line 100, in force_text
    s = s.__unicode__()
  File "/vagrant/expo/scheduler/models.py", line 342, in __unicode__
    return unicode(self.describe)
  File "/vagrant/expo/scheduler/models.py", line 336, in describe
    resourceitem_id=self.resourceitem_id).name
  File "/usr/local/lib/python2.7/dist-packages/model_utils/managers.py", line 185, in get_subclass
    return self.get_queryset().get_subclass(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/model_utils/managers.py", line 160, in get_subclass
    return self.select_subclasses().get(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/query.py", line 310, in get
    self.model._meta.object_name)
DoesNotExist: LocationItem matching query does not exist.
```

Not the exact same error.

But similar.

This only happens when I do "runcoverage".  I'm running it on GBE-711, and not master... but since this isn't an area where there is change, I will not be surprised if master is similar.

There is some ick around how Rooms are setup.  I had a failing test case I couldn't fix months ago connected to not being able to retrieve a setup LocationItem as well... there's something creepy there

**Comment by jonkiparsky** [comment]
There are some notes on this on the (closed) #731

**Comment by bethlakshmi** [comment]
@jonkiparsky - do we think this is obsolete with pytest?
If you agree, let's close it.
